docs/extend/middleware.md

```diff
@@ -1 +1,109 @@
-#  Middleware
\ No newline at end of file
+#  Middleware
+
+Middleware is a nifty way to wrap the handling of HTTP requests in Flarum. This can allow you to modify responses, add your own checks to the request, and much more. The possibilities are endless!
+
+```php
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class YourMiddleware implements MiddlewareInterface {
+  public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+  {    
+    // Your logic here!
+    return $handler->handle($request);
+  }
+}
+```
+
+## Adding Middleware In Your Extension
+
+To add a new middleware, simply use the middleware extender in your extension's `extend.php` file:
+
+```php
+use Flarum\Extend;
+
+return [
+  // Add middleware to forum frontend
+  (new Extend\Middleware('forum'))->add(YourMiddleware::class),
+  // Admin frontend
+  (new Extend\Middleware('admin'))->add(YourMiddleware::class),
+  // API frontend
+  (new Extend\Middleware('api'))->add(YourMiddleware::class)
+];
+```
+
+Tada! Middleware defined.
+
+Now that we've got the basics down, let's run through a few more things:
+
+## Restricting Middleware to Certain Routes
+
+If you don't need your middleware to execute under every route, you can add an `if` to filter it:
+
+```php
+use Laminas\Diactoros\Uri;
+
+$currentRoute = $request->getUri()->getPath();
+$routeToRunUnder = new Uri(app()->url('/path/to/run/under'));
+
+if ($currentRoute === $routeToRunUnder->getPath()) {
+  // Your logic here!
+}
+```
```

Comment:
Keen to hear your (and other readers') opinion here, but I think this might be confusing without at least the skeleton of the `handle()` method. So, IMO, at least that function body should always be included. 

If we want to keep the example short, we can avoid the `use` lines by using fully-qualified class names everywhere. But that's a tradeoff...